lib/tapioca/dsl/compiler.rb

```diff
@@ -44,20 +46,51 @@ def processable_constants
           )
         end
 
+        sig { params(constants: T::Array[Module]).void }
+        def requested_constants=(constants)
+          @@requested_constants = constants
+        end
+
         private
 
+        sig do
+          type_parameters(:U)
+            .params(klass: T.all(T::Class[T.anything], T.type_parameter(:U)))
+            .returns(T::Array[T.type_parameter(:U)])
+        end
+        def descendants_of(klass)
+          if @@requested_constants.any?
+            T.cast(
+              @@requested_constants.select do |k|
+                k < klass && !k.singleton_class?
+              end,
```

Comment:
I'm not sure this will be sufficient.

Imagine we're editing a superclass, and we make changes that'll effect the RBIs for the subclasses. Those subclasses will need to change, but they might not be in our requested constants (only the superclass would be)

Reply:
What you're describing is a possibility with requested constants as whole. This PR is just making existing functionality faster. We filter the constants coming from `gather_constants` to match `requested_constants` [here](https://github.com/Shopify/tapioca/blob/main/lib/tapioca/dsl/pipeline.rb#L156).

I think you're coming from the LSP Addon perspective and yes if we want it to cover all cases it'd have to be smarter at identifying which constants to generate. But that's not the scope of this PR.